Jetpack Backup: Redirect back to main backup page when clicking on `Back up now` button

Initsogar opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Resolves https://github.com/Automattic/jetpack-backup-team/issues/482

## Proposed Changes
* Add a custom `onClick` handler on `BackupNowButton`, that will give us more flexibility on adding custom logic when clicking that button. 
* Redirect to the main backup page when clicking on `Back up now` button, so the user can monitor the progress of the backup being created

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Currently, if the user clicks on Back up now, it indicates their interest in securing their latest changes. Therefore, we want to ensure they can monitor the progress of the backup being created.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Spin up a **Calypso Blue** or **Jetpack Cloud** live branch
* Pick a site with Jetpack VaultPress Backup plan with more than 1 backup
* Navigate to the backup page
* Pick a date on the calendar rather than Today
* Click on `Back up now`
* You should be redirected back to `Today` and you should see the progress of the backup being created.

To validate how it was working before, you can try the same steps in production.



## Demo

https://github.com/Automattic/wp-calypso/assets/1488641/4165f1d3-7d9d-4624-a0ed-29bc3848d058

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Initsogar committed:
Redirect back to main backup page when clicking on `Back up now` button

Initsogar committed:
Refactor handler for clicking BackupNowButton

Initsogar committed:
Remove siteSlug that is no longer needed in BackupNowButton

Initsogar commented on client/components/jetpack/backup-now-button/index.tsx:
```diff
@@ -81,15 +89,13 @@ const BackupNowButton: FunctionComponent< Props > = ( {
 		}
 	}, [ areBackupsStopped, backupCurrentlyInProgress, tooltipText, translate, enqueued, children ] );
 
-	// TODO: If we want to re-enable the backup button we can set enqueue to false once we detect a backup in progress
```
This comment is no longer necessary.

Initsogar committed:
Remove onClick default value as it is an optional prop

mavegaf commented:
Tested with Calypso and with Jetpack Cloud. It seems it only works in Jetpack Cloud, Is that expected

Initsogar commented:
> Tested with Calypso and with Jetpack Cloud. It seems it only works in Jetpack Cloud, Is that expected

@mavegaf Hmm, it should work on both. I'll take a look. Thanks!

Initsogar committed:
Refactor backup actions toolbar

Removing siteSlug as a prop, and getting it from a selector inside the component. Add the logic to decide if we should show the `Copy site` button and add the onBackupNowClick inside this component.

Initsogar committed:
Update backup actions toolbar README file to reflect that siteSlug is no longer needed

Initsogar committed:
Use BackupActionsToolbar component for Calypso Blue and Green.

Initsogar commented on client/my-sites/backup/main.jsx:
```diff
@@ -106,13 +105,7 @@ const BackupPage = ( { queryDate } ) => {
-						>
-							{ translate( 'Back up now' ) }
-						</BackupNowButton>
+						<BackupActionsToolbar siteId={ siteId } />
```
Now we are going to use `BackupActionsToolbar` for Calypso Blue and Green.

Initsogar commented on client/my-sites/backup/main.jsx:
```diff
@@ -214,11 +206,7 @@ function BackupStatus( { selectedDate, needCredentials, onDateChange } ) {
-									<BackupActionsToolbar siteId={ siteId } siteSlug={ siteSlug } />
-								</>
-							) }
+							<BackupActionsToolbar siteId={ siteId } />
```
`siteSlug` validation is no longer needed, as we refactored the component to query the siteSlug directly.

Initsogar commented on client/components/jetpack/backup-actions-toolbar/index.tsx:
```diff
@@ -1,25 +1,31 @@
 	const hasBackups = useSelector( ( state ) => siteHasBackups( state, siteId ) );
 
+	// Show the "Copy site" button if accessing on Jetpack Cloud or A4A
+	const showCopySiteButton = isJetpackCloud() || isA8CForAgencies();
```
This logic was outside this component. I think it is better to use BackupActionsToolbar everywhere, so we don't need to duplicate code. Before we were rendering `BackupNowButton` directly when the user is navigating through Jetpack Cloud.

Initsogar commented:
> Tested with Calypso and with Jetpack Cloud. It seems it only works in Jetpack Cloud, Is that expected

@mavegaf fixed! I did forget that we were rendering only the `BackupNowButton` directly when the user is navigating on Calypso Blue. So I refactored the `BackupActionsToolbar` in a way that we could use that component for both cases, and inside we decide if we need to display or not the `Copy site` button.

Could you take another look? :)

mavegaf reviewed (approved):
Tested in both, no errors in console 👍
